File: Examples/CSharp/ProgrammersGuide/WorkingWithProjects/RetrieveCalendarInfo.cs
using System.IO;
using Aspose.Tasks;
using System.Collections;
using System;
using System.Collections.Generic;

namespace Aspose.Tasks.Examples.CSharp.WorkingWithProjects
{
    public class RetrieveCalendarInfo
    {
        public static void Run()
        {
            //ExStart: RetrieveCalendarInfo
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Projects();

            //Create a project reader instance
            Project project = new Project(dataDir + "RetrieveCalendarInfo.mpp");

            //Retrieve Calendars Information
            CalendarCollection alCals = project.Calendars;
            foreach (Aspose.Tasks.Calendar cal in alCals)
            {
                if (cal.Name != null)
                {
                    Console.WriteLine("Calendar UID : " + cal.Uid);
                    Console.WriteLine("Calendar Name : " + cal.Name);

                    WeekDayCollection alDays = cal.WeekDays;
                    foreach (WeekDay wd in alDays)
                    {
                        TimeSpan ts = wd.GetWorkingTime();
                        if (wd.DayWorking)
                        {
                            Console.WriteLine(wd.DayType.ToString() + ":");
                            Console.WriteLine(ts.ToString());
                        }
                    }
                }
            }
            //ExEnd: RetrieveCalendarInfo
        }
    }
}

File: Examples/VisualBasic/ProgrammersGuide/WorkingWithProjects/RetrieveCalendarInfo.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Tasks
Imports System.Collections
Imports System
Imports System.Collections.Generic

Namespace Aspose.Tasks.Examples.VisualBasic.WorkingWithProjects
    Public Class RetrieveCalendarInfo

        Public Shared Sub Run()
            'ExStart: RetrieveCalendarInfo
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Projects()

            'Create a project reader instance
            Dim project As New Project(dataDir & "RetrieveCalendarInfo.mpp")

            'Retrieve Calendars Information
            Dim alCals As CalendarCollection = project.Calendars
            For Each cal As Aspose.Tasks.Calendar In alCals
                If cal.Name IsNot Nothing Then
                    Console.WriteLine("Calendar UID : " & cal.Uid)
                    Console.WriteLine("Calendar Name : " & cal.Name)

                    Dim alDays As WeekDayCollection = cal.WeekDays
                    For Each wd As WeekDay In alDays
                        Dim ts As TimeSpan = wd.GetWorkingTime()
                        If wd.DayWorking Then
                            Console.WriteLine(wd.DayType.ToString() & ":")
                            Console.WriteLine(ts.ToString())
                        End If
                    Next wd
                End If
            Next cal
            'ExEnd: RetrieveCalendarInfo
        End Sub
    End Class
End Namespace